.github/workflows/generator_container_slsa3.yml

```diff
@@ -195,6 +209,50 @@ jobs:
 
           echo "login to ${untrusted_registry}"
           cosign login "${untrusted_registry}" -u "${username}" -p "${password}"
+      
+      - name: Provenance registry login
+        id: provenance-registry-login
+        continue-on-error: true
+        env:
+          UNTRUSTED_PROVENANCE_REGISTRY: "${{ inputs.provenance-registry  }}"
+          UNTRUSTED_SECRET_PROVENANCE_REGISTRY: "${{ secrets.provenance-registry }}"
+          UNTRUSTED_INPUT_PROVENANCE_USERNAME: "${{ inputs.provenance-registry-username }}"
+          UNTRUSTED_SECRET_PROVENANCE_USERNAME: "${{ secrets.provenance-registry-username }}"
+          UNTRUSTED_PROVENANCE_PASSWORD: "${{ secrets.provenance-password }}"
+          GCP_ACCESS_TOKEN: "${{ steps.auth.outputs.access_token }}"
+        if: ${{ env.UNTRUSTED_SECRET_PROVENANCE_REGISTRY != '' || env.UNTRUSTED_PROVENANCE_REGISTRY != '' }}
+        run: |
+          set -euo pipefail
+
+          # NOTE: Some registries are of the form <org>/<name>
```

Comment:
Let's add a TODO(xxx) to consolidate this code into a script, since it's a duplicate of the code used for signing into the main registry. Can you create a tracking issue for it?

Reply:
#2974

Reply:
Added todo